src/main/java/Duke.java

```diff
@@ -1,43 +1,96 @@
+        for (int i = 2; i < inputWords.length; i++) {
+            sb.append(inputWords[i]);
+            if (i < inputWords.length - 1) {
+                sb.append(" ");
+            }
+        }
+        return sb.toString();
+    }
+
+    private static void makeTaskDone(String[] inputWords, Task[] tasks) {
+        for (int i = 1; i < inputWords.length; i++) {
+            try {
+                int num = Integer.parseInt(inputWords[i]) - 1;
+                tasks[num].markAsDone();
+                System.out.println("Congrats! You've completed this task: [X] "
+                        + tasks[num].getTaskDesc());
+            } catch (NumberFormatException e) {
+                System.out.println(SOMETHING_WRONG +
+                        "Please ensure that you have entered the index of the task(s) to be marked as done");
+                break;
+            }
+        }
+    }
+
+    private static void listAllTasks(Task[] tasks) {
+        for (int i = 0; i < nextAdd; i++) {
+//            StringBuilder sb = new StringBuilder();
+            String done = tasks[i].isPending() ? "[ ]" : "[X]";
+            System.out.println((i + 1) + ": " + tasks[i].getType() + done + " "
+                    + tasks[i].getTaskDesc() + " " + tasks[i].getAdditionalInfo());
+        }
+    }
+
+
     private static void userCommands() {
 
         boolean isProgramRunning = true;
         Scanner scanner = new Scanner(System.in);
         Task[] tasks = new Task[100];
-        int nextAdd = 0;
 
         while (isProgramRunning) {
             System.out.print("Enter your command here: ");
             String input = scanner.nextLine();
-            if (input.contains("done")) {
-                String[] words = input.split(" ");
-                for (int i = 1; i < words.length; i++) {
-                    int num = Integer.parseInt(words[i]) - 1;
-                    tasks[num].markAsDone();
-                    System.out.println("Congrats! You've completed this task: [X] "
-                            + tasks[num].getTaskDesc());
-                }
+            if (input.isEmpty()) {
+                //TODO: print something wrong
+            } else if (input.equals("help")) {
+                //TODO: print help statement
             } else if (input.equals("bye")) {
                 isProgramRunning = false;
             } else if (input.equals("list")) {
-                for (int i = 0; i < nextAdd; i++) {
-                    String done = tasks[i].isPending() ? "[ ]" : "[X]";
-                    System.out.println((i + 1) + ": " + done + " " + tasks[i].getTaskDesc());
-                }
+                listAllTasks(tasks);
             } else {
-                Task temp = new Task(input);
-                tasks[nextAdd] = temp;
-                System.out.println("Added: " + input);
-                nextAdd++;
+                String[] inputWords = input.split(" ");
+                if (inputWords[0].equals("done")) {
```

Comment:
better use a constant to define all the commands like “done”, “deadline”, “todo”, etc.